src/v/datalake/coordinator/coordinator_manager.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#pragma once
+
+#include "cluster/fwd.h"
+#include "cluster/notification.h"
+#include "datalake/coordinator/coordinator.h"
+#include "model/fundamental.h"
+#include "raft/fwd.h"
+#include "raft/notification.h"
+
+#include <seastar/core/abort_source.hh>
+#include <seastar/core/future.hh>
+#include <seastar/core/gate.hh>
+#include <seastar/core/sharded.hh>
+
+namespace datalake::coordinator {
+
+// Manages the lifecycle of datalake coordinators, each of which operate on a
+// single partition of the control topic.
+class coordinator_manager {
+public:
+    coordinator_manager(
+      model::node_id self,
+      ss::sharded<raft::group_manager>&,
+      ss::sharded<cluster::partition_manager>&);
+
+    ss::future<> start();
+    ss::future<> stop();
+
+    ss::lw_shared_ptr<coordinator> get(const model::ntp&) const;
+
+private:
+    void start_managing(cluster::partition&);
+    void stop_managing(const model::ntp&);
+    void notify_leadership_change(
+      raft::group_id group,
+      model::term_id term,
+      std::optional<model::node_id> leader_id);
+
+    model::node_id self_;
```

Comment:
nit: The usual pattern I see in the code is _self  (_ as a prefix).

Reply:
Yea some newer areas of the codebase have this (e.g. src/v/io) but agreed definitely not common. I kinda prefer it from the GSG, though happy to change it at some point module-wide if it's causing problems